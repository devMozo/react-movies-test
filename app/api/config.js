export const URL_MOVIE_API = 'https://cors-anywhere.herokuapp.com/https://api.themoviedb.org/3';
