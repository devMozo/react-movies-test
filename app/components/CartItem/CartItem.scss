.CartItem {
    list-style: none;
    margin-top: 20px;

    a {
        width: 500px;
        display: inline-block;
    }

    img {
        width: 100px;
        height: 100px;
        margin-right: 20px;
        object-fit: cover;
        vertical-align: middle;
    }

    .CartItem__price {
        width: 200px;
        text-align: right;
        color: #333;
    }

    .CartItem__remove {
        color: white;
        background-color: #ff3d00;
        padding: 10px 30px;
        margin-left: 20px;
        border: 0;
        transition: 0.3s;
        cursor: pointer;

        &:hover {
            opacity: 0.6;
        }
    }
}
