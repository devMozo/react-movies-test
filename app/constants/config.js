export const MOVIES_API_KEY = '1038f4f1dddddb86a86db5baf8cfaf04';
export const MOVIES_IMG_PATH = 'https://image.tmdb.org/t/p/w500/';
