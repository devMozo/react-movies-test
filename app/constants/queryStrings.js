export const QUERY_API_KEY = 'api_key';
