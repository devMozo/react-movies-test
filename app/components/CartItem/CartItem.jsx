import React from 'react';
import { MOVIES_IMG_PATH } from '../../constants/config';
import { Link } from 'react-router-dom';
import './CartItem.scss';

const CartItem = ({ item, onRemove }) => {
    return (
        <li className="CartItem">
            <picture>
                <img src={`${MOVIES_IMG_PATH}${item.poster_path}`} />
            </picture>
            <Link className="CartItem__title" to={`movie/${item.id}`}>
                {item.title}
            </Link>
            <b className="CartItem__price"> ${item.budget} </b>
            <button className="CartItem__remove" onClick={() => onRemove(item.id)}>
                {' '}
                Remove{' '}
            </button>
        </li>
    );
};

export default React.memo(CartItem);
