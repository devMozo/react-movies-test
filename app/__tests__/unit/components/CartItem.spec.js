import React from 'react';
import { shallow } from 'enzyme';
import CartItem from '../../../components/CartItem/CartItem';
import { MOVIES_IMG_PATH } from '../../../constants/config';

describe('<CartItem />', () => {
    const mockedOnRemove = jest.fn();
    const PROPS = {
        item: {
            poster_path: 'mocked_poster',
            vote_average: 2,
            title: 'mocked_title',
            overview: 'mocked_overview',
            production_companies: [
                {
                    name: 'Company 1',
                },
                {
                    name: 'Company 2',
                },
            ],
            production_countries: [
                {
                    name: 'Country 1',
                },
                {
                    name: 'Country 2',
                },
            ],
            status: 'Released',
            budget: 123456789,
            adult: false,
            release_date: '17-2-2019',
        },
        onRemove: mockedOnRemove,
    };
    let component;

    beforeEach(() => {
        component = shallow(<CartItem {...PROPS} />);
    });

    it('should render correctly in "debug" mode', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render correctly each field', () => {
        const image = component.find('.CartItem img');
        const title = component.find('.CartItem__title');
        const price = component.find('.CartItem__price');

        expect(image.prop('src')).toBe(`${MOVIES_IMG_PATH}${PROPS.item.poster_path}`);
        expect(title.contains(PROPS.item.title)).toBeTruthy();
        expect(price.contains(PROPS.item.budget)).toBeTruthy();
    });

    it('should trigger the onRemove action when the user hits the button to remove', () => {
        const button = component.find('.CartItem__remove');
        button.simulate('click');

        expect(mockedOnRemove).toHaveBeenCalled();
    });
});
