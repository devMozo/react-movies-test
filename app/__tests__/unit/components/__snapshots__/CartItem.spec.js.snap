// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should render correctly in "debug" mode 1`] = `
<li
  className="CartItem"
>
  <picture>
    <img
      src="https://image.tmdb.org/t/p/w500/mocked_poster"
    />
  </picture>
  <Link
    className="CartItem__title"
    to="movie/undefined"
  >
    mocked_title
  </Link>
  <b
    className="CartItem__price"
  >
     $
    123456789
     
  </b>
  <button
    className="CartItem__remove"
    onClick={[Function]}
  >
     
    Remove
     
  </button>
</li>
`;
